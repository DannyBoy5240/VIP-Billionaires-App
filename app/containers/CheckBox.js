import React from 'react'
import { Text, TouchableOpacity } from 'react-native'
import Ionicons from 'react-native-vector-icons/Ionicons'
import { COLOR_BLACK_LIGHT, COLOR_BLUE } from '../constants/colors'

const CheckBox = React.memo(
  ({
     title,
     checked,
     checkedColor,
     unCheckedColor,
     checkedIcon,
     uncheckedIcon,
     textStyle,
     containerStyle,
     onPress,
   }) => (
    <TouchableOpacity
      style={{
        ...containerStyle,
        flexDirection: 'row',
        alignItems: 'center',
        height: 49,
        paddingHorizontal: 12,
        borderWidth: 1,
        borderRadius: 11,
        borderColor: checked ? COLOR_BLUE : COLOR_BLACK_LIGHT,
        marginBottom: 5,
      }}
      onPress={onPress}>
      <Ionicons
        name={checked ? checkedIcon : uncheckedIcon}
        size={12}
        color={checked ? checkedColor : unCheckedColor}
      />
      {title ? (
        <Text style={{ marginLeft: 8, ...textStyle }}>{title}</Text>
      ) : null}
    </TouchableOpacity>
  ),
)
export default CheckBox
