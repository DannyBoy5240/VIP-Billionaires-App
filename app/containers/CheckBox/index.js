import React from 'react'
import { Text, TouchableOpacity } from 'react-native'
import { COLOR_YELLOW } from '../../constants/colors'
import { VectorIcon } from '../VectorIcon'

const Index = React.memo(
  ({ title, checked, textStyle, containerStyle, onPress }) => (
    <TouchableOpacity
      style={{
        ...containerStyle,
        flexDirection: 'row',
        alignItems: 'center',
        paddingVertical: 8,
      }}
      onPress={onPress}>
      <VectorIcon
        name={checked ? 'check-square-o' : 'square-o'}
        type={'FontAwesome'}
        size={24}
        color={checked ? COLOR_YELLOW : 'black'}
      />
      {title ? (
        <Text style={{ marginLeft: 8, ...textStyle }}>{title}</Text>
      ) : null}
    </TouchableOpacity>
  ),
)

export default Index
