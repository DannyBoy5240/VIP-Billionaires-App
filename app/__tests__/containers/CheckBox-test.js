import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react-native';
import CheckBox from '../../containers/CheckBox';

test('renders correctly', () => {
  const tree = renderer.create(<CheckBox />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('checkbox performance', () => {
  const mockFn = jest.fn();

  render(<CheckBox title="CHECKBOX" onPress={mockFn} />);
  expect(screen.getByText('CHECKBOX')).toBeTruthy();
  
  fireEvent.press(screen.getByText('CHECKBOX'));
  expect(mockFn).toBeCalledWith();
});