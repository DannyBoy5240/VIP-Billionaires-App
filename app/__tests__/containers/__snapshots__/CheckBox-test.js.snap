// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": undefined,
      "borderRadius": 11,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 49,
      "marginBottom": 5,
      "opacity": 1,
      "paddingHorizontal": 12,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        undefined,
        Object {
          "fontFamily": undefined,
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  />
</View>
`;
